File: src/main/leetcode/100. Same Tree.kt
import lecture062_Tree_Introduction.TreeNode

/*
*   https://leetcode.com/problems/same-tree/
* */
private fun isSameTree(p: TreeNode?, q: TreeNode?): Boolean {

    // Base case
    if (p == null && q == null) {
        return true
    }
    if (p == null && q != null) {
        return false
    }
    if (p != null && q == null) {
        return false
    }

    val left = isSameTree(p!!.left, q!!.left)
    val right = isSameTree(p.right, q.right)
    val isSameData = p.data == q.data

    return left && right && isSameData
}

File: src/main/leetcode/100. Same Tree.cpp
import lecture062_Tree_Introduction.TreeNode

/*
*   https://leetcode.com/problems/same-tree/
* */

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

private:
    bool isSameTree(TreeNode* p, TreeNode* q) {

        // Base case
        if (p == NULL && q == NULL) {
            return true;
        }
        if (p == NULL && q != NULL) {
            return false;
        }
        if (p != NULL && q == NULL) {
            return false;
        }

        bool left = isSameTree(p->left, q->left);
        bool right = isSameTree(p->right, q->right);
        bool isSameData = p->val == q->val;

        return left && right && isSameData;
    }


